File: TextosJapones/src/main/java/org/jisho/textosJapones/fileparse/Parse.java
package org.jisho.textosJapones.fileparse;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.util.List;
import java.util.Map;

public interface Parse {
	void parse(File arquivo) throws IOException;

	void destroi() throws IOException;

	String getTipo();

	InputStream getPagina(int numero) throws IOException;

	int getSize();

	List<String> getLegenda();

	Map<String, Integer> getLegendaNomes();

	String getPaginaPasta(Integer num);

	Map<String, Integer> getPastas();
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/fileparse/Parse.kt
package br.com.fenix.processatexto.fileparse

import java.io.File
import java.io.IOException
import java.io.InputStream

interface Parse {

    @Throws(IOException::class)
    fun parse(arquivo: File)

    @Throws(IOException::class)
    fun destroir()

    fun getTipo(): String

    @Throws(IOException::class)
    fun getPagina(numero: Int): InputStream

    fun getSize(): Int

    fun getLegenda(): List<String>

    fun getLegendaNomes(): Map<String, Int>

    fun getPaginaPasta(num: Int): String

    fun getPastas(): Map<String, Int>

}